### PR TITLE
[Feature] [Plugin] add spark cache transform

### PR DESCRIPTION
**Pull request by lizu18xz**
add spark cache transform

### CODE REVIEW

File: docs/en/transform/cache.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+DISK_ONLY_2
+MEMORY_ONLY
+MEMORY_ONLY_2
+MEMORY_ONLY_SER
+MEMORY_ONLY_SER_2
+MEMORY_AND_DISK
+MEMORY_AND_DISK_2
+MEMORY_AND_DISK_SER
+MEMORY_AND_DISK_SER_2
+OFF_HEAP
```
Comment (ashulin) [inline docs/en/transform/cache.md]: You can describe their differences, or add a reference link to Spark

Comment (lizu18xz) [reply docs/en/transform/cache.md]: ok，I improve the document

File: seatunnel-connectors/seatunnel-connectors-spark/pom.xml [hunk 2]
```diff
@@ -94,6 +100,11 @@
             <groupId>org.apache.spark</groupId>
             <artifactId>spark-sql_${scala.binary.version}</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.fasterxml.jackson.module</groupId>
+            <artifactId>jackson-module-scala_2.11</artifactId>
+            <version>2.12.6</version>
```
Comment (Hisoka-X) [inline seatunnel-connectors/seatunnel-connectors-spark/pom.xml]: version should use parmeter, like ${jackson.scala.module.version}

**Review by hailin0 — COMMENTED** [review]
LGTM

**Review by ashulin — COMMENTED** [review]
Can you add test case to any IT in `seatunnel-spark-e2e` module

**Comment by lizu18xz** [comment]
> Can you add test case to any IT in `seatunnel-spark-e2e` module

such as  FakeSourceToConsoleIT？ add to this class?

**Review by Hisoka-X — DISMISSED** [review]
LGTM, Please waiting PR passed.

**Comment by Hisoka-X** [comment]
Can you merge code from dev to retry ci?

**Comment by lizu18xz** [comment]
> 

ok

**Review by Hisoka-X — APPROVED** [review]
LGTM, Thanks for your contribute and long follow up！

**Review by EricJoy2048 — APPROVED** [review]

**Pull request merged**